dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/datasource/repository/CaseSubmissionRepository.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.laa.crime.dces.integration.datasource.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.justice.laa.crime.dces.integration.datasource.model.CaseSubmissionEntity;
+
+import java.math.BigInteger;
+
+@Repository
+public interface CaseSubmissionRepository extends JpaRepository<CaseSubmissionEntity, Integer> {
+
+    @Query(value = "select nextval('trace_id_seq')", nativeQuery = true)
+    BigInteger getNextTraceId();
+
+    @Query(value = "select nextval('batch_id_seq')", nativeQuery = true)
+    BigInteger getNextBatchId();
```

Comment:
Do these need to be `BigInteger`? A primitive `long` would be much easier to deal with in the code, and we're not going to reach `Long.MAX_VALUE` (9_223_372_036_854_775_807) anytime soon (and you have int/Integer id values elsewhere anyway, which would run out first).

Reply:
I agree it's overkill, however it was to keep the use of BigInteger across the application, at least for the more recent things.
The changes to the .xsd have caused all of those previously Integer/etc values to become BigIntegers, which is adding to the fun.

I think a backlog task to unify all these might be an idea truth be told. For example maatId to a Long, instead of a BigInteger, a Long, and an Integer all in the same app. 

What's your thoughts?

Reply:
For the task for aligning all these, I have created 
https://dsdmoj.atlassian.net/browse/DCES-560
Happy to pick it up later myself.